### ISSUE TITLE
Conda osx-arm64 release?

### ISSUE THREAD

**Issue by nhadler** [hjkgrp/molSimplify#170]
I wanted to know if it was planned to release a version of molsimplify on conda-forge with support for osx-arm64? As of right now there is only linux-64 and  osx-64. #42 suggests that arm mac's should be able to run molsimplify.

**Comment by gianmarco-terrones** [comment]
Hi, we will look into this in our next molSimplify release, which will happen soon.

**Comment by gianmarco-terrones** [comment]
https://anaconda.org/hjkgroup/molsimplify/files
Actually, it seems there is a release with osx-arm64

**Comment by nhadler** [comment]
huh, that's odd that my conda install isn't finding it. Let me try again

**Comment by nhadler** [comment]
Okay so it looks like there's two releases of molsimplifly---one on conda-forge (https://anaconda.org/conda-forge/molsimplify) and one at the hjkgroup repo (https://anaconda.org/hjkgroup/molsimplify). The former doesn't have the arm release? I think it would be worth adding a link to the hjkgroup in the github readme.

**Comment by gianmarco-terrones** [comment]
I have updated the readme in commit #772135ae4f5e4df642424959a479c2c49a24d179

**Comment by gianmarco-terrones** [comment]
Regarding the conda forge version with python 3.10, do all of the pytest tests pass?

**Comment by nhadler** [comment]
Outside of some warnings, it passed the full pytest suite. I'm going to rebuild the environment with 3.11 and see if it passes, there's a chance for a decent speed up using that release.

<details><summary>pytest logs</summary>
<p>
```bash
====================================================================================================== test session starts =======================================================================================================
platform linux -- Python 3.10.12, pytest-7.4.2, pluggy-1.3.0
rootdir: /home/nhadler/projects/molSimplify
plugins: anyio-4.0.0
collected 590 items                                                                                                                                                                                                              

tests/test_basic_imports.py .s.s.                                                                                                                                                                                          [  0%]
tests/test_bidentate.py .                                                                                                                                                                                                  [  1%]
tests/test_distgeom.py ....                                                                                                                                                                                                [  1%]
tests/test_example_1.py ..                                                                                                                                                                                                 [  2%]
tests/test_example_3.py s.                                                                                                                                                                                                 [  2%]
tests/test_example_5.py ..                                                                                                                                                                                                 [  2%]
tests/test_example_7.py ..                                                                                                                                                                                                 [  3%]
tests/test_example_8.py .                                                                                                                                                                                                  [  3%]
tests/test_example_tbp.py .                                                                                                                                                                                                [  3%]
tests/test_ff_xtb.py .s....                                                                                                                                                                      [  4%]
tests/test_geocheck_oct.py ................                                                                                                                                                      [  7%]
tests/test_geocheck_one_empty.py ..                                                                                                                                                              [  7%]
tests/test_inparse.py .....                                                                                                                                                                      [  8%]
tests/test_io.py .                                                                                                                                                                               [  8%]
tests/test_jobgen.py ....                                                                                                                                                                        [  9%]
tests/test_joption_pythonic.py .                                                                                                                                                                 [  9%]
tests/test_ligand_assign_consistent.py ..                                                                                                                                                        [  9%]
tests/test_ligand_from_mol_file.py s                                                                                                                                                             [  9%]
tests/test_ligands.py .................................................................................................................................................................................................... [ 43%]
.......................................................................................................................................................................................................................... [ 80%]
..............................................................                                                                                                                                   [ 90%]
tests/test_mol3D.py ...s............                                                                                                                                                             [ 93%]
tests/test_molcas_caspt2.py .                                                                                                                                                                    [ 93%]
tests/test_molcas_casscf.py .                                                                                                                                                                    [ 93%]
tests/test_old_ANNs.py ..                                                                                                                                                                        [ 93%]
tests/test_orca_ccsdt.py .                                                                                                                                                                       [ 94%]
tests/test_orca_dft.py .                                                                                                                                                                         [ 94%]
tests/test_qcgen.py ......                                                                                                                                                                       [ 95%]
tests/test_racs.py .....s.                                                                                                                                                                       [ 96%]
tests/test_rmsd.py ....                                                                                                                                                                          [ 97%]
tests/test_structgen_functions.py ....                                                                                                                                                           [ 97%]
tests/test_tetrahedral.py ..                                                                                                                                                                     [ 98%]
tests/test_tutorial_09_part_one.py .                                                                                                                                                             [ 98%]
tests/test_tutorial_09_part_two.py .                                                                                                                                                             [ 98%]
tests/test_tutorial_10_part_one.py .                                                                                                                                                             [ 98%]
tests/test_tutorial_10_part_two.py .                                                                                                                                                             [ 98%]
tests/test_tutorial_2.py .                                                                                                                                                                       [ 98%]
tests/test_tutorial_3.py .                                                                                                                                                                       [ 99%]
tests/test_tutorial_6.py .                                                                                                                                                                       [ 99%]
tests/test_tutorial_8.py ..                                                                                                                                                                      [ 99%]
tests/test_tutorial_qm9_part_one.py .                                                                                                                                                            [ 99%]
tests/informatics/test_coulomb_analyze.py .                                                                                                                                                      [100%]

=========================================================================================== warnings summary ===========================================================================================
molSimplify/python_nn/ensemble_test.py:11
  /home/nhadler/projects/molSimplify/molSimplify/python_nn/ensemble_test.py:11: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    from pkg_resources import resource_filename, Requirement

../../mambaforge/envs/molsimp/lib/python3.10/site-packages/pkg_resources/__init__.py:2871
  /home/nhadler/mambaforge/envs/molsimp/lib/python3.10/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

../../mambaforge/envs/molsimp/lib/python3.10/site-packages/pkg_resources/__init__.py:2871
  /home/nhadler/mambaforge/envs/molsimp/lib/python3.10/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('google')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

tests/test_bidentate.py: 10 warnings
tests/test_example_1.py: 3 warnings
tests/test_example_5.py: 2 warnings
tests/test_example_7.py: 2 warnings
tests/test_ff_xtb.py: 5 warnings
tests/test_molcas_caspt2.py: 1 warning
tests/test_molcas_casscf.py: 1 warning
tests/test_orca_ccsdt.py: 1 warning
tests/test_orca_dft.py: 1 warning
tests/test_structgen_functions.py: 1 warning
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/tf_nn_prep.py:716: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    print(f"ANN predicts a spin splitting (HS - LS) of {float(split[0]):.2f} kcal/mol at {100 * alpha:.0f}% HFX")

tests/test_bidentate.py: 10 warnings
tests/test_example_1.py: 3 warnings
tests/test_example_5.py: 2 warnings
tests/test_example_7.py: 2 warnings
tests/test_ff_xtb.py: 5 warnings
tests/test_molcas_caspt2.py: 1 warning
tests/test_molcas_casscf.py: 1 warning
tests/test_orca_ccsdt.py: 1 warning
tests/test_orca_dft.py: 1 warning
tests/test_structgen_functions.py: 1 warning
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/tf_nn_prep.py:723: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    print(f"ANN predicts a HOMO value of {float(homo[0]):.2f} eV at {100 * alpha:.0f}% HFX")

tests/test_bidentate.py: 10 warnings
tests/test_example_1.py: 3 warnings
tests/test_example_5.py: 2 warnings
tests/test_example_7.py: 2 warnings
tests/test_ff_xtb.py: 5 warnings
tests/test_molcas_caspt2.py: 1 warning
tests/test_molcas_casscf.py: 1 warning
tests/test_orca_ccsdt.py: 1 warning
tests/test_orca_dft.py: 1 warning
tests/test_structgen_functions.py: 1 warning
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/tf_nn_prep.py:724: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    print(f"ANN predicts a LUMO-HOMO energetic gap value of {float(gap[0]):.2f} eV at {100 * alpha:.0f}% HFX")

tests/test_mol3D.py::test_finding_and_counting_methods
  /home/nhadler/projects/molSimplify/molSimplify/utils/decorators.py:13: DeprecationWarning: Duplicate function will be removed in a future release. Use findAtomsbySymbol instead.
    warnings.warn(message, DeprecationWarning)

tests/test_old_ANNs.py::test_old_ANN
tests/test_structgen_functions.py::test_init_ANN
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/nn_prep.py:462: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    "{0:.2f}".format(float(r)) + ' angstrom'))

tests/test_old_ANNs.py::test_old_ANN
tests/test_structgen_functions.py::test_init_ANN
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/nn_prep.py:465: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    "{0:.2f}".format(float(r_ls)) + ' angstrom'))

tests/test_old_ANNs.py::test_old_ANN
tests/test_structgen_functions.py::test_init_ANN
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/nn_prep.py:467: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    "{0:.2f}".format(float(r_hs)) + ' angstrom'))

tests/test_old_ANNs.py::test_old_ANN
tests/test_structgen_functions.py::test_init_ANN
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/nn_prep.py:473: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    "{0:.2f}".format(float(HFX_slope)) + ' kcal/HFX'))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================= 583 passed, 7 skipped, 93 warnings in 611.62s (0:10:11) ========================================================================
```
</p>
</details>

**Comment by nhadler** [comment]
It passes with 3.11 as well. It would be great if the pinned dependency on the conda could be updated.

<details><summary>pytest with Python 3.11</summary>
<p>

```
============================= test session starts ==============================
platform linux -- Python 3.11.5, pytest-7.4.2, pluggy-1.3.0
rootdir: /home/nhadler/projects/molSimplify
plugins: anyio-4.0.0
collected 590 items

tests/test_basic_imports.py .s.s.                                        [  0%]
tests/test_bidentate.py .                                                [  1%]
tests/test_distgeom.py ....                                              [  1%]
tests/test_example_1.py ..                                               [  2%]
tests/test_example_3.py s.                                               [  2%]
tests/test_example_5.py ..                                               [  2%]
tests/test_example_7.py ..                                               [  3%]
tests/test_example_8.py .                                                [  3%]
tests/test_example_tbp.py .                                              [  3%]
tests/test_ff_xtb.py .s....                                              [  4%]
tests/test_geocheck_oct.py ................                              [  7%]
tests/test_geocheck_one_empty.py ..                                      [  7%]
tests/test_inparse.py .....                                              [  8%]
tests/test_io.py .                                                       [  8%]
tests/test_jobgen.py ....                                                [  9%]
tests/test_joption_pythonic.py .                                         [  9%]
tests/test_ligand_assign_consistent.py ..                                [  9%]
tests/test_ligand_from_mol_file.py s                                     [  9%]
tests/test_ligands.py .................................................. [ 18%]
........................................................................ [ 30%]
........................................................................ [ 42%]
........................................................................ [ 54%]
........................................................................ [ 67%]
........................................................................ [ 79%]
..................................................................       [ 90%]
tests/test_mol3D.py ...s............                                     [ 93%]
tests/test_molcas_caspt2.py .                                            [ 93%]
tests/test_molcas_casscf.py .                                            [ 93%]
tests/test_old_ANNs.py ..                                                [ 93%]
tests/test_orca_ccsdt.py .                                               [ 94%]
tests/test_orca_dft.py .                                                 [ 94%]
tests/test_qcgen.py ......                                               [ 95%]
tests/test_racs.py .....s.                                               [ 96%]
tests/test_rmsd.py ....                                                  [ 97%]
tests/test_structgen_functions.py ....                                   [ 97%]
tests/test_tetrahedral.py ..                                             [ 98%]
tests/test_tutorial_09_part_one.py .                                     [ 98%]
tests/test_tutorial_09_part_two.py .                                     [ 98%]
tests/test_tutorial_10_part_one.py .                                     [ 98%]
tests/test_tutorial_10_part_two.py .                                     [ 98%]
tests/test_tutorial_2.py .                                               [ 98%]
tests/test_tutorial_3.py .                                               [ 99%]
tests/test_tutorial_6.py .                                               [ 99%]
tests/test_tutorial_8.py ..                                              [ 99%]
tests/test_tutorial_qm9_part_one.py .                                    [ 99%]
tests/informatics/test_coulomb_analyze.py .                              [100%]

=============================== warnings summary ===============================
../../mambaforge/envs/molsimp/lib/python3.11/site-packages/tensorflow/python/debug/cli/debugger_cli_common.py:19
  /home/nhadler/mambaforge/envs/molsimp/lib/python3.11/site-packages/tensorflow/python/debug/cli/debugger_cli_common.py:19: DeprecationWarning: module 'sre_constants' is deprecated
    import sre_constants

molSimplify/python_nn/ensemble_test.py:11
  /home/nhadler/projects/molSimplify/molSimplify/python_nn/ensemble_test.py:11: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    from pkg_resources import resource_filename, Requirement

../../mambaforge/envs/molsimp/lib/python3.11/site-packages/pkg_resources/__init__.py:2871
  /home/nhadler/mambaforge/envs/molsimp/lib/python3.11/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

../../mambaforge/envs/molsimp/lib/python3.11/site-packages/pkg_resources/__init__.py:2871
  /home/nhadler/mambaforge/envs/molsimp/lib/python3.11/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('google')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

molSimplify/Classes/mWidgets.py:18
  /home/nhadler/projects/molSimplify/molSimplify/Classes/mWidgets.py:18: DeprecationWarning: 'imghdr' is deprecated and slated for removal in Python 3.13
    import imghdr

tests/test_bidentate.py: 10 warnings
tests/test_example_1.py: 3 warnings
tests/test_example_5.py: 2 warnings
tests/test_example_7.py: 2 warnings
tests/test_ff_xtb.py: 5 warnings
tests/test_molcas_caspt2.py: 1 warning
tests/test_molcas_casscf.py: 1 warning
tests/test_orca_ccsdt.py: 1 warning
tests/test_orca_dft.py: 1 warning
tests/test_structgen_functions.py: 1 warning
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/tf_nn_prep.py:716: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    print(f"ANN predicts a spin splitting (HS - LS) of {float(split[0]):.2f} kcal/mol at {100 * alpha:.0f}% HFX")

tests/test_bidentate.py: 10 warnings
tests/test_example_1.py: 3 warnings
tests/test_example_5.py: 2 warnings
tests/test_example_7.py: 2 warnings
tests/test_ff_xtb.py: 5 warnings
tests/test_molcas_caspt2.py: 1 warning
tests/test_molcas_casscf.py: 1 warning
tests/test_orca_ccsdt.py: 1 warning
tests/test_orca_dft.py: 1 warning
tests/test_structgen_functions.py: 1 warning
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/tf_nn_prep.py:723: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    print(f"ANN predicts a HOMO value of {float(homo[0]):.2f} eV at {100 * alpha:.0f}% HFX")

tests/test_bidentate.py: 10 warnings
tests/test_example_1.py: 3 warnings
tests/test_example_5.py: 2 warnings
tests/test_example_7.py: 2 warnings
tests/test_ff_xtb.py: 5 warnings
tests/test_molcas_caspt2.py: 1 warning
tests/test_molcas_casscf.py: 1 warning
tests/test_orca_ccsdt.py: 1 warning
tests/test_orca_dft.py: 1 warning
tests/test_structgen_functions.py: 1 warning
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/tf_nn_prep.py:724: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    print(f"ANN predicts a LUMO-HOMO energetic gap value of {float(gap[0]):.2f} eV at {100 * alpha:.0f}% HFX")

tests/test_mol3D.py::test_finding_and_counting_methods
  /home/nhadler/projects/molSimplify/molSimplify/utils/decorators.py:13: DeprecationWarning: Duplicate function will be removed in a future release. Use findAtomsbySymbol instead.
    warnings.warn(message, DeprecationWarning)

tests/test_old_ANNs.py::test_old_ANN
tests/test_structgen_functions.py::test_init_ANN
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/nn_prep.py:462: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    "{0:.2f}".format(float(r)) + ' angstrom'))

tests/test_old_ANNs.py::test_old_ANN
tests/test_structgen_functions.py::test_init_ANN
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/nn_prep.py:465: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    "{0:.2f}".format(float(r_ls)) + ' angstrom'))

tests/test_old_ANNs.py::test_old_ANN
tests/test_structgen_functions.py::test_init_ANN
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/nn_prep.py:467: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    "{0:.2f}".format(float(r_hs)) + ' angstrom'))

tests/test_old_ANNs.py::test_old_ANN
tests/test_structgen_functions.py::test_init_ANN
  /home/nhadler/projects/molSimplify/molSimplify/Scripts/nn_prep.py:473: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    "{0:.2f}".format(float(HFX_slope)) + ' kcal/HFX'))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========== 583 passed, 7 skipped, 95 warnings in 657.28s (0:10:57) ============

```

</p>
</details>

**Comment by gianmarco-terrones** [comment]
Thank you for your help @nhadler. Pinned dependency on conda repo, regarding python 3.11, has been updated. Please see https://anaconda.org/hjkgroup/molsimplify. Let us know if that works for you

**Comment by nhadler** [comment]
Thanks @gianmarco-terrones! It appears to be working for me on linux. Would it be possible to loosen the pinned requirements to 3.10.* as well?
